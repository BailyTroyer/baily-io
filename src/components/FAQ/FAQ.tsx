import React, { FC } from "react"

interface Props {
  question: string
  answer: string
}

const FAQ: FC<Props> = (props: Props) => {
  const { question, answer } = props

  const [visible, setVisible] = React.useState(false)

  return (
    <div
      className="mx-auto my-3 flex flex-col max-w-3xl cursor-pointer select-none overflow-hidden"
      onClick={() => setVisible(!visible)}
    >
      <div className="bg-purple-400 rounded-t-md px-6 py-3">
        <h1 className="font-medium text-white lg:text-2xl text-xl">
          {question}
        </h1>
      </div>
      <div
        className={`flex-col bg-purple-100 px-5 py-2 ${
          visible ? "" : "hidden"
        }`}
      >
        <p>{answer}</p>
      </div>
    </div>
  )
}

export default FAQ
