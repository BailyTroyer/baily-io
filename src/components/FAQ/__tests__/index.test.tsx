import React from "react"

import renderer from "react-test-renderer"

import FAQ from "../FAQ"

it("renders correctly", () => {
  const tree = renderer
    .create(<FAQ question="QUETION" answer="ANSWER" />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
