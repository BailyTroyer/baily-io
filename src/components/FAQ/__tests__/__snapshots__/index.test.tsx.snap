// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="mx-auto my-3 flex flex-col max-w-3xl cursor-pointer select-none overflow-hidden"
  onClick={[Function]}
>
  <div
    className="bg-purple-400 rounded-t-md px-6 py-3"
  >
    <h1
      className="font-medium text-white lg:text-2xl text-xl"
    >
      QUETION
    </h1>
  </div>
  <div
    className="flex-col bg-purple-100 px-5 py-2 hidden"
  >
    <p>
      ANSWER
    </p>
  </div>
</div>
`;
